source/plugins/ruby/extension_utils.rb

```diff
@@ -109,5 +108,35 @@ def getNamespaceFilteringModeForDataCollection
       end
       return namespaceFilteringMode
     end
+
+    def getDataCollectionProfile
+      begin
+        dataCollectionStreamProfile = "Default"
+        dataTypes = []
+        dataTypeToStreamIdMap = Extension.instance.get_stream_mapping()
+        if !dataTypeToStreamIdMap.nil? && !dataTypeToStreamIdMap.empty?
+          dataTypes = dataTypeToStreamIdMap.keys
+          if (dataTypes.length >= 12)
+            dataCollectionStreamProfile = "Default"
+          elsif ((dataTypes.length <= 3) && (dataTypes.include?("CONTAINER_LOG_BLOB") || dataTypes.include?("CONTAINERINSIGHTS_CONTAINERLOGV2")) && dataTypes.include?("KUBE_EVENTS_BLOB"))
+            dataCollectionStreamProfile = "LogsAndEvents"

```

Comment:
Hey @ganga1980 just confirming, through the UX we allow the customers to pick multiple stream profiles (for example they could pick "LogsAndEvents" and "PV". Looks like if that were the case, the dataCollectionStreamProfile will simply get set to "LogsAndEvents"--will that cause any problems further in the code? I haven't done a full review just saw this snippet in my email and wanted to confirm.

Reply:
> Hey @ganga1980 just confirming, through the UX we allow the customers to pick multiple stream profiles (for example they could pick "LogsAndEvents" and "PV". Looks like if that were the case, the dataCollectionStreamProfile will simply get set to "LogsAndEvents"--will that cause any problems further in the code? I haven't done a full review just saw this snippet in my email and wanted to confirm.

> @sarahpeiffer , No impact to the functionality and agent supports the scenario one or more profiles or steams. This code is just for the telemetry. If customer chooses more than one profile or picks the specific streams, then agent side, telemetry captures as "custom" profile.